Introduction for the manuscript (#12)
greenelab/text_mined_hetnet_manuscript: content/02.introduction.md

# Introduction

> 

Reviewer: Would start with: why do we care about knowledge bases? Are there some cool examples of how they get used?

Reviewer: Addressed by the new sentences 👍

> Populating knowledge bases is a daunting and arduous task that takes a significant amount of manual effort to accomplish.

Reviewer: Is there any literature on this with specifics?

> Populating knowledge bases is a daunting and arduous task that takes a significant amount of manual effort to accomplish.

Reviewer: ```suggestion
Populating knowledge bases often requires highly-trained scientists to read biomedical literature and summarize the results.
This manual curation process requires a significant amount of time and effort.
```

Reviewer: I suggested a way to reframe these sentence, but the second sentence would really benefit from some specific figures provided by a reference/references if available.

> Typically, this task involves  manual curation, which is the process of curators reading and parsing published papers to extract relationships of interest. 

Reviewer: ```suggestion
```

> This framework becomes infeasible as the rate of publications continues to increase exponentially [@doi:10.1002/asi.23329].

Reviewer: ```suggestion
The rate of publications is increasing exponentially [@doi:10.1002/asi.23329], but the number of curators is not.
We expect this will lead to gaps in knowledge bases.
```

> A common solution towards this problem is to undergo relationship extraction.

Reviewer 2: ```suggestion
A common solution towards this problem is to perform relationship extraction.
```

undergo has a passive connotation: "experience or be subjected to"

> A common solution towards this problem is to perform relationship extraction.

Reviewer: ```suggestion
```

> Relationship extraction is the process of creating a machine learning system to automatically scan and extract relationships from textual sources.

Reviewer: Has relationship extraction been proposed as a solution to this problem? It seems like someone must have proposed this in the literature in the past. If so, this should be re-worked to include that reference here.

> Relationship extraction is the process of creating a machine learning system to automatically scan and extract relationships from textual sources.

Reviewer: ```suggestion

Relationship extraction is the process of creating a machine learning system to automatically scan and extract relationships from textual sources.
```

Also, this seems like a new paragraph.

> The caveat towards this solution is that these systems need a non-trivial amount of training examples to work well.

Reviewer: ```suggestion
Machine learning methods often require a large corpus of well-labeled training data to succeed, which still requires manual curation.
```

> To generate these examples involves the arduous process of hand-labeling each example one by one.  

Reviewer: ```suggestion
```

> A common technique to ease this process is to use distant supervision, which makes the assumption that that all sentences, that contain an entity pair found in a well established database, will provide evidence for a desired relationship [@doi:10.3115/1690219.1690287].

Reviewer: ```suggestion
Distant supervision is one technique to sidestep the requirement of well-annotated sentences: with distant supervision one makes the assumption that that all sentences containing an entity pair found in a selected database provide evidence for a relationship [@doi:10.3115/1690219.1690287].
```

> This technique is useful for quickly generating training examples; however, it generates a lot of noise i.e. a signficiant increase in the number of false positives.  

Reviewer: ```suggestion
Distant supervision provides many labeled examples; however it is accompanied by a decrease in the quality of the labels.  
```

> Recently, there has been work that builds off of this idea called the data programming paradigm [@arxiv:1605.07723].

Reviewer: ```suggestion
Ratner et al. [@arxiv:1605.07723] recently introduced "data programming" as a solution.
Data programming combines distant supervision with the automated labeling of text using hand-written label functions.
The distant supervision sources and label functions are integrated using a noise aware generative model, which is used to produce training labels.
```

> This paradigm generates large annotated datasets quickly by using weak supervision strategies in the form of label functions and train a noise aware model to incorporate these strategies.

Reviewer 2: "by using weak supervision strategies in the form of label functions and train a noise"

some sort of grammer issue

> This paradigm generates large annotated datasets quickly by using weak supervision strategies in the form of label functions and train a noise aware model to incorporate these strategies.

Reviewer: ```suggestion
```

> Empirically, creating a useful label function under this paradigm can take a significant amount of time.

Reviewer: ```suggestion
Combining distant supervision with label functions can dramatically reduce the time required to acquire sufficient training data.
However, constructing a knowledge base of heterogeneous relationships through this framework still requires tens of hand-written label functions for each relationship type.
Writing useful label functions requires significant error analysis, and can be time-consuming.
```

> In this paper, we aim to answer the question: can we accelerate the label function generation process by reusing label functions across different relationship types?

Reviewer: ```suggestion

In this paper, we aim to address the question: to what extent can we re-use label functions across different relationship types?
(a brief 1-2 sentence blurb about why this is plausible)
Label function re-use could dramatically reduce the number of label functions required to generate a heterogeneous knowledge graph, which would save substantial researcher time.
```

## Introduction

Knowledge bases are important resources that hold complex structured and unstructed information. 
These resources have been used in important tasks such as network analysis for drug repurposing discovery [@doi:10.1371/journal.pone.0084912; @doi:10.1101/385617; @doi:10.7554/eLife.26726] or as a source of training labels for text mining systems [@doi:10.3115/1690219.1690287; @doi:10.1101/444398; @doi:10.1186/s12859-019-2873-7]. 
Populating knowledge bases often requires highly-trained scientists to read biomedical literature and summarize the results [@doi:10.1093/bib/bbn043].

> This manual curation process requires a significant amount of effort and time (approx 8.4 years on average [@doi:10.1093/bioinformatics/btm229]).

Reviewer: ```suggestion
This manual curation process requires a significant amount of effort and time: in 2007 researchers estimated that filling in the missing annotations to that point would require approximately 8.4 years [@doi:10.1093/bioinformatics/btm229]).
```

> The rate of publications is increasing exponentially [@doi:10.1002/asi.23329], but the number of curators is not.

Reviewer: ```suggestion
The rate of publications has continued to increase exponentially [@doi:10.1002/asi.23329].
```

Reviewer: Changed this one since the previous pub was from 2007

> We expect this will lead to gaps in knowledge bases.  

Reviewer: ```suggestion
This has been recognized as a considerable challenge and leads to gaps in knowledge bases [@doi:10.1093/bioinformatics/btm229].  
```

Relationship extraction has been studied as a solution towards handling this problem [@doi:10.1093/bib/bbn043].
This process consists of creating a machine learning system to automatically scan and extract relationships from textual sources.

> Machine learning methods often require a large corpus of well-labeled training data to succeed, which still requires manual curation.

Reviewer: ```suggestion
Machine learning methods often leverage a large corpus of well-labeled training data, which still requires manual curation.
```

Reviewer: require was 2x in the same sentence

Distant supervision is one technique to sidestep the requirement of well-annotated sentences: with distant supervision one makes the assumption that that all sentences containing an entity pair found in a selected database provide evidence for a relationship [@doi:10.3115/1690219.1690287].
Distant supervision provides many labeled examples; however it is accompanied by a decrease in the quality of the labels.  
Ratner et al. [@arxiv:1605.07723] recently introduced "data programming" as a solution.
Data programming combines distant supervision with the automated labeling of text using hand-written label functions.
The distant supervision sources and label functions are integrated using a noise aware generative model, which is used to produce training labels.
Combining distant supervision with label functions can dramatically reduce the time required to acquire sufficient training data.
However, constructing a knowledge base of heterogeneous relationships through this framework still requires tens of hand-written label functions for each relationship type.
Writing useful label functions requires significant error analysis, which can be a time-consuming process.  

> In this paper, we aim to address the question: to what extent can we re-use label functions across different relationship types?

Reviewer: ```suggestion

In this paper, we aim to address the question: to what extent can label functions be re-used across different relationship types?
```

> We hypothesize that text for one relationship may share information in the form of keywords or sentence structure with text for other relationship types.

Reviewer: ```suggestion
We hypothesized that sentences describing one relationship type may share information in the form of keywords or sentence structure with sentences that indicate other relationship types.
```

> We test this hypothesis via label function re-use.

Reviewer: ```suggestion
We designed a series of experiments to determine the extent to which label function re-use enhanced performance over distant supervision alone.
We examine relationships that indicate similar types of physical interactions (i.e., gene-binds-gene and compound-binds-gene) as well as different types (i.e., disease-associates-gene and compound-treats-disease).
```

> Through re-use we could dramatically reduce the number of label functions required to generate a heterogeneous knowledge graph, which would save substantial researcher time.

Reviewer: ```suggestion
The re-use of label functions could dramatically reduce the number required to generate and update a heterogeneous knowledge graph.
```
